rfd/0109-opsgenie-plugin.md

```diff
@@ -0,0 +1,110 @@
+# Opsgenie Integration RFD
+## Required approvers
+
+Engineering:
+Product:
+Security:
+
+## What
+
+This RFD proposes a plugin that allows Teleport to integrate with Opsgenie, allowing access requests to show up as alerts in Opsgenie. This plugin will differ from existing plugins by being part of the Teleport binary directly.
+
+## Success criteria
+Users are able to configure teleport to automatically create alerts in Opsgenie from access requests.
+Alerts were chosen over incidents in Opsgenie as incidents are intended to be used for high priority alerts that indicate a service interruption.
+Users are also able to configure auto approval flows to be met under certain conditions. E.g when a requester is on-call 
+
+## Configuration UX
+
+The plugin will be configured in Teleport's config yaml file. The required fields will be added to a 'plugins' section containing the required information to interact with both Teleport access and the Opsgenie API.
+
+```
+plugins:
+    opsgenie:
+        api_key: "path/to/key" # File containing Opsgenie API Key
+        opsgenie_addr: "example.app.opsgenie.com" # Address of Opsgenie
+        priority: "2" # Priority to create Opsgenie alerts with
+        alert_tags: ["example-tag"] # List of tags to be added to alerts created in Opsgenie
+        auto_approval: true # Whether or not to enable auto approval
+```
+
+The logging configuration will be shared with the main Teleport process.
+
+### Getting an Opsgenie API key
+
+In the Opsgenie web UI go to Settings -> App settings -> API key management. Create key with Read, Create and Update access.
+
+### Executing
+
+The plugin will be started using a command of the form
+
+```
+teleport start --config /etc/teleport.yaml
+```
+
+## UX
+
+Once an access request has been created, the Opsgenie plugin will create an alert in the service specified in the request annotation using the Opsgenie Alert API Create endpoint. 
+
+The appropriate on call responder can then click into the provided link and approve or deny the access request.
+
+For auto approval of certain access requests the access request will be auto approved if the requesting user is on-call in one of the services provided in request annotation. (Provided auto approval is enabled in the config)
+
+Once an access request has been approved or denied the plugin will add a note to the alert and close the relevant alert tied to that access request.
+
+## Implementation details
```

Comment:
In the implementation details, can you include a section that will explain how the plugin will be integrated in the teleport binary? Are we going to import stuff from teleport-plugins? I think @justinas mentioned we already do this for the hosted plugins work?